video2dataset/data_reader.py

```diff
@@ -123,37 +150,52 @@ def __call__(self, url):
             yt_meta_dict = get_yt_meta(url, self.metadata_args)
         else:
             yt_meta_dict = None, None
-        return path, yt_meta_dict, None
+
+        audio_path = None
+        if self.encode_formats.get("audio", None):
+            af = self.encode_formats["audio"]
+            audio_path = video2audio(path, af, self.sample_rate, self.tmp_dir)
+        if not self.encode_formats.get("video", None):
+            os.remove(path)
+            path = None
+        return path, audio_path, yt_meta_dict, None
 
 
 class VideoDataReader:
     """Video data reader provide data for a video"""
 
-    def __init__(self, video_size, dl_timeout, tmp_dir, yt_meta_args) -> None:
-        self.mp4_downloader = Mp4Downloader(dl_timeout, tmp_dir)
-        self.yt_downloader = YtDlpDownloader(tmp_dir, yt_meta_args, video_size)
+    def __init__(self, video_size, dl_timeout, tmp_dir, yt_meta_args, encode_formats) -> None:
+        self.mp4_downloader = Mp4Downloader(dl_timeout, tmp_dir, encode_formats)
+        self.yt_downloader = YtDlpDownloader(tmp_dir, yt_meta_args, video_size, encode_formats)
 
     def __call__(self, row):
         key, url = row
 
         yt_meta_dict = None
+        a_file_path = None
+        aud_bytes = None
+        vid_bytes = None
         # TODO: make nice function to detect what type of link we're dealing with
         if "youtube" in url:  # youtube link
             try:
-                file_path, yt_meta_dict, error_message = self.yt_downloader(url)
+                file_path, a_file_path, yt_meta_dict, error_message = self.yt_downloader(url)
             except Exception as e:  # pylint: disable=(broad-except)
-                file_path, yt_meta_dict, error_message = None, None, str(e)
+                file_path, a_file_path, yt_meta_dict, error_message = None, None, None, str(e)
         # TODO: add .avi, .webm, should also work
         elif url.endswith(".mp4"):  # mp4 link
-            file_path, error_message = self.mp4_downloader(url)
+            file_path, a_file_path, error_message = self.mp4_downloader(url)
         else:
-            file_path, error_message = None, "Warning: Unsupported URL type"
+            file_path, a_file_path, error_message = None, None, "Warning: Unsupported URL type"
 
         if error_message is None:
-            with open(file_path, "rb") as vid_file:
-                vid_bytes = vid_file.read()
+            if file_path is not None:
+                with open(file_path, "rb") as vid_file:
+                    vid_bytes = vid_file.read()
+            if a_file_path is not None:
+                with open(a_file_path, "rb") as aud_file:
+                    aud_bytes = aud_file.read()
         else:
             vid_bytes = None
         if file_path is not None:  # manually remove tempfile
             os.remove(file_path)
-        return key, vid_bytes, yt_meta_dict, error_message
+        return key, vid_bytes, aud_bytes, yt_meta_dict, error_message
```

Comment:
can we return dict instead of video and audio bytes separately. Just make a dict with same keys as encode_formats but value is the bytes